File: sdks/python/examples/basic_query.py
"""
Basic query — send a prompt and collect the result.

Run: uv run python examples/basic_query.py
"""
import asyncio
import sys

from forgecode import query


async def main() -> None:
    async for message in query("What is 2 + 2? Reply with just the number."):
        match message.type:
            case "system":
                print(f"[session] {message.session_id}")
            case "assistant":
                sys.stdout.write(message.content)
            case "result":
                print(f"\n[result] {message.result}")
            case "error":
                print(f"[error] {message.error}", file=sys.stderr)


if __name__ == "__main__":
    asyncio.run(main())

File: sdks/typescript/examples/basic-query.ts
/**
 * Basic query — send a prompt and collect the result.
 *
 * Run: bun run examples/basic-query.ts
 */
import { query } from "../src";

for await (const message of query({
  prompt: "What is 2 + 2? Reply with just the number.",
})) {
  switch (message.type) {
    case "system":
      console.log(`[session] ${message.session_id}`);
      break;
    case "assistant":
      process.stdout.write(message.content);
      break;
    case "result":
      console.log(`\n[result] ${message.result}`);
      break;
    case "error":
      console.error(`[error] ${message.error}`);
      break;
  }
}
